### DISCUSSION TITLE
Stop all motors when one fader is touched

### DISCUSSION THREAD

**Discussion by Vahen1981** [tttapa/Control-Surface-Motor-Fader#5] [Q&A]
Hi Pieter!
Sometimes I need to move more than one fader at the same time, so I tried to modify your code to get this... I found this lines:  

```if (Config::enable_controller) {
        if (touched){ // Turn off motor if knob is touched
            motors.setSpeed<Idx>(0);
        }
        else
            motors.setSpeed<Idx>(control);
    }
```
So I changed to some like this:

```if (Config::enable_controller) {
        if (touched){ // Turn off motor if knob is touched
            motors.setSpeed<0>(0);
            motors.setSpeed<1>(0);
            motors.setSpeed<2>(0);
            motors.setSpeed<3>(0);
        }
        else
            motors.setSpeed<Idx>(control);
    }
```

But, it doesn't happens what I want... Do you have some suggestion about what should I do?

**Comment by tttapa** [comment]
I think it's easiest to change these lines:
https://github.com/tttapa/Control-Surface-Motor-Fader/blob/9b20111e7eb25f09a17da3a1a4eb6438c0d54247/Motor-Controller/main.cpp#L286-L287

You could use something like this:
```cpp
    // Check if any of the fader knobs is touched
    bool touched = false;
    for (uint8_t i = 0; i < Config::num_faders; ++i)
        touched |= touch.read(i);
```
